pkg/miscellaneous/miscellaneous.go

```diff
@@ -539,14 +539,32 @@ func ReadConfigFileAsMap(path string) (map[string]interface{}, error) {
 	return c, nil
 }
 
-// ParsePeerURL forms a PeerURL, given podName by parsing the initial-advertise-peer-urls
-func ParsePeerURL(initialAdvertisePeerURLs, podName string) (string, error) {
-	tokens := strings.Split(initialAdvertisePeerURLs, "@")
-	if len(tokens) < 4 {
-		return "", fmt.Errorf("invalid peer URL : %s", initialAdvertisePeerURLs)
+// ParseAdvertiseURL parses the advertise URL and returns the protocol, service name, namespace and port.
+func ParseAdvertiseURL(advUrl string) (string, string, string, string, error) {
```

Comment:
If https://github.com/gardener/etcd-backup-restore/pull/715#discussion_r1774884199 makes sense to you, then you can go ahead and make this helper function `parseAdvertiseURL`, or simply make it a part of the function body of `CreateMemberPeerURL` instead of seperating it out into its own function. Either works for me. You can also just leave it as an exported function if you would prefer that.